optaplanner-persistence/optaplanner-persistence-jackson/src/test/java/org/optaplanner/persistence/jackson/api/score/constraint/IndictmentJacksonJsonSerializerAndDeserializerTest.java

```diff
@@ -0,0 +1,97 @@
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.Before;
+import org.junit.Test;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.core.api.score.constraint.Indictment;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+import static org.junit.Assert.assertEquals;
+
+public class IndictmentJacksonJsonSerializerAndDeserializerTest {
+
+    ObjectMapper mapper;
+
+    @Before
+    public void setUp() throws Exception {
+        mapper = new ObjectMapper();
+        mapper.registerModule(new SimpleModule()
+                .addSerializer(ConstraintMatch.class, new ConstraintMatchJacksonJsonSerializer())
+                .addDeserializer(ConstraintMatch.class, new ConstraintMatchJacksonJsonDeserializer())
+                .addSerializer(HardSoftScore.class, new ScoreJacksonJsonSerializer<HardSoftScore>())
+                .addDeserializer(HardSoftScore.class, new HardSoftScoreJacksonJsonDeserializer()));
+    }
+
+    private void testSerializeAndDeserialize(Object cause, Score zeroScore, ConstraintMatch... constraints) {
+        Indictment indictment = new Indictment(cause, zeroScore);
+        for (ConstraintMatch constraint : constraints) {
+            indictment.addConstraintMatch(constraint);
+        }
+        TestIndictmentWrapper wrapper = new TestIndictmentWrapper(indictment);
+
+        try {
+            String json = mapper.writeValueAsString(wrapper);
+
+            TestIndictmentWrapper other = mapper.readValue(json, TestIndictmentWrapper.class);
+
+            assertEquals(indictment, other.getIndictment());
+        } catch (JsonParseException e) {
+            throw new IllegalStateException(e);
+        } catch (JsonMappingException e) {
+            throw new IllegalStateException(e);
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Test
+    public void testSerializeAndDeserialize() {
+        testSerializeAndDeserialize("Hello", S(0, 0));
+        testSerializeAndDeserialize(1, S(0, 1), CM("package", "name", 0, 10, "Bad"));
+        testSerializeAndDeserialize("What", S(0, 1), CM("package", "name", 0, 10, "Bad"), CM("package", "b", 10, 20,
+                "Very Bad"));
+    }
+
+    private static HardSoftScore S(int h, int s) {
+        return HardSoftScore.valueOf(h, s);
```

Comment:
Is this method worth it's weight in characters written?

Reply:
In this example no, since `CM(...)` calls it implicitly. In `ConstraintMatchTotalJacksonJsonSerializerAndDeserializerTest`, it is (otherwise line 66 would be much longer). If I make `CM(...)` accept a score rather than 2 integers, then it'll be worth its weight (plus we can test multiple score types).